python_tests/test_python_frontend.py

```diff
@@ -1468,5 +1469,48 @@ def fusion_func(fd: FusionDefinition) :
             self.assertEqual(torch.real(inputs[0]), nvf_out[0])
             self.assertEqual(torch.imag(inputs[0]), nvf_out[1])
 
+    def test_pad(self):
+        inputs = [
+            torch.testing.make_tensor((2, 3), dtype=torch.float32, device="cuda"),
+            2,  # used as symbolic pad width input
+        ]
+
+        def fusion_func(fd: FusionDefinition):
+            t0 = fd.from_pytorch(inputs[0])
+
+            # Need to do stuff like this until we merge
+            # https://github.com/csarofeen/pytorch/pull/2449
+            one = fd.define_constant(1)
+
+            t1 = fd.ops.pad(t0, [one, one, one, one])
+            fd.add_output(t1)
+
+            t2 = t0.pad([one, one, one, one])
+            fd.add_output(t2)
+
+            # no padding in some dims (see note above)
+            zero = fd.define_constant(0)
+            two = fd.define_constant(2)
+            three = fd.define_constant(3)
+            t3 = fd.ops.pad(t0, [zero, zero, two, three])
+            fd.add_output(t3)
+
+            # no padding in all dims
+            t4 = fd.ops.pad(t0, [zero, zero, zero, zero])
+            fd.add_output(t4)
+
+            # passing symbolic pad widths
+            s0 = fd.define_scalar(DataType.Int)
+            t5 = t0.pad([s0, s0, s0, s0])
+            fd.add_output(t5)
```

Comment:
can't we handle runtime input for pad? We should test that!

Reply:
I thought no, but maybe it should just work? Actual padding widths affect tensor sizes, but as long as scheduler heuristics don't change, I think the same scheduled kernel should work.

Same for `slice`. For `cat`, the dimension to concatenate is currently assumed to be static (i.e., `int` not `Val`). It should be possible to make it dynamic too.

Reply:
Just to clarify, looking at the API and Naoya's comment above, we should test it!

If dynamic padding size is not supported, maybe we should rethink our python API to make it clear to user.

Reply:
@jjsjann123 sorry for late reply. I may be misunderstanding here; the lines you quoted above are actually using a dynamic padding size. That scalar `s0` is passed as an input.